pandas/tseries/index.py

```diff
@@ -833,7 +833,7 @@ def astype(self, dtype):
         if dtype == np.object_:
             return self.asobject
         elif dtype == _INT64_DTYPE:
```

Comment:
hmm, this is tricky, the problem is that ``NaT`` is defined as the - max value for int64. So the point is to ALWAYS convert to i8, no matter the integer input (e.g. even ``.astype('i2')`` would be returned as i8.